src/views/darkroom.c

```diff
@@ -539,6 +539,13 @@ void expose(dt_view_t *self,
   {
     // draw image
     _view_paint_surface(cri, width, height, port, DT_WINDOW_MAIN);
+    if(dt_conf_get_bool("darkroom/ui/loading_screen") == FALSE) {
+      // cache the rendered bitmap for use while loading the next image
+      if(darktable.gui->surface)
```

Comment:
I'd suggest moving this outside the `if(dt_conf_get_bool("darkroom/ui/loading_screen")` (and setting `gui->surface = NULL`) so that toggling on the loading screen releases the memory.